libraries/SD_MMC/src/SD_MMC.cpp

```diff
@@ -57,7 +59,9 @@ SDMMCFS::SDMMCFS(FSImplPtr impl) : FS(impl), _card(nullptr) {
   _pin_d3 = SDMMC_SLOT1_IOMUX_PIN_NUM_D3;
 #endif  // BOARD_HAS_1BIT_SDMMC
 
-#elif SOC_SDMMC_USE_IOMUX && defined(BOARD_HAS_SDMMC) && defined(CONFIG_IDF_TARGET_ESP32P4)
+// ESP32-P4 can use either IOMUX or GPIO matrix
+#elif defined(BOARD_HAS_SDMMC) && defined(CONFIG_IDF_TARGET_ESP32P4)
+#if defined(BOARD_SDMMC_SLOT) && (BOARD_SDMMC_SLOT == 0)
   _pin_clk = SDMMC_SLOT0_IOMUX_PIN_NUM_CLK;
```

Comment:
Does this actually work? IIUC, the current sdmmc driver expects all pins to be set to 0 for IOMUX slots.

Reply:
We need the pins for Peripheral manager.
Later the config is cleared, as the SDMMC_SLOT_CONFIG_DEFAULT for P4 is setting the pins in the config.
```
sdmmc_slot_config_t slot_config = SDMMC_SLOT_CONFIG_DEFAULT();

#if defined(CONFIG_IDF_TARGET_ESP32P4) && defined(BOARD_SDMMC_SLOT) && (BOARD_SDMMC_SLOT == 0)
  host.slot = SDMMC_HOST_SLOT_0;
  // reconfigure slot_config to remove all pins in order to use IO_MUX
  slot_config = {
    .cd = SDMMC_SLOT_NO_CD,
    .wp = SDMMC_SLOT_NO_WP,
    .width = 4,
    .flags = 0,
  };
#else
```

Reply:
For P4 it will be always this config:
```
#define SDMMC_SLOT_CONFIG_DEFAULT() {\
    .clk = GPIO_NUM_43, \
    .cmd = GPIO_NUM_44, \
    .d0 = GPIO_NUM_39, \
    .d1 = GPIO_NUM_40, \
    .d2 = GPIO_NUM_41, \
    .d3 = GPIO_NUM_42, \
    .d4 = GPIO_NUM_45, \
    .d5 = GPIO_NUM_46, \
    .d6 = GPIO_NUM_47, \
    .d7 = GPIO_NUM_48, \
    .cd = SDMMC_SLOT_NO_CD, \
    .wp = SDMMC_SLOT_NO_WP, \
    .width   = SDMMC_SLOT_WIDTH_DEFAULT, \
    .flags = 0, \
}
```

Reply:
As both IOMUX and GPIO Matrix are supported, this will never be used for P4.

#if SOC_SDMMC_USE_IOMUX && !SOC_SDMMC_USE_GPIO_MATRIX
```
/**
 * Macro defining default configuration of SDMMC host slot
 */
#define SDMMC_SLOT_CONFIG_DEFAULT() {\
    .cd = SDMMC_SLOT_NO_CD, \
    .wp = SDMMC_SLOT_NO_WP, \
    .width   = SDMMC_SLOT_WIDTH_DEFAULT, \
    .flags = 0, \
}
```